openmdao/solvers/linear/linear_schur.py

```diff
@@ -0,0 +1,381 @@
+        self._delta_d_n_1 = None
+        self._mode_linear = mode_linear
+        self._sys_names = sys_names
+
+    def _declare_options(self):
+        """
+        Declare options before kwargs are processed in the init method.
+        """
+        super()._declare_options()
+        # this solver does not iterate
+        self.options.undeclare('maxiter')
+        self.options.undeclare('err_on_non_converge')
+
+        self.options.undeclare('atol')
+        self.options.undeclare('rtol')
+
+    def _setup_solvers(self, system, depth):
+        """
+        Assign system instance, set depth, and optionally perform setup.
+
+        Parameters
+        ----------
+        system : System
+            pointer to the owning system.
+        depth : int
+            depth of the current system (already incremented).
+        """
+        super()._setup_solvers(system, depth)
+
+        if self._sys_names is None:
+            self._sys_names = [s for s in system._subsystems_allprocs.keys() if s != '_auto_ivc']
+
+        if len(self._sys_names) != 2:
+            raise ValueError(f'System {self.pathname} has a LinearSchur solver and is required to '
+                             'contain two subsystems, but it has '
+                             f'{len(self._sys_names)}.\n{self._sys_names}')
+
+    def solve(self, mode, rel_systems=None):
+        """
+        Perform the operations in the iteration loop.
+
+        Parameters
+        ----------
+        mode : str
+            Specifies whether the solver is operating in forward ('fwd') or reverse ('rev') mode.
+        rel_systems : set of str
+            Set of names of relevant systems based on the current linear solve.  Deprecated.
+        """
+        self._mode = mode
+        self._rel_systems = rel_systems
+        system = self._system()
+        mode = self._mode_linear
+
+        if self._mode == 'fwd':
+            self._rhs_vec = system._dresiduals.asarray(True)
+        else:
+            self._rhs_vec = system._doutputs.asarray(True)
+
+        # take the subsystems
+        subsys1, _ = system._subsystems_allprocs[self._sys_names[0]]
+        subsys2, _ = system._subsystems_allprocs[self._sys_names[1]]
+
+        # TODO this may not be the most general case. think about just solving for a subset
+        subsys2_outputs = subsys2._doutputs
+        subsys2_residuals = subsys2._dresiduals
+
+        # list of variables we solve for here. this should include all variables in
+        # subsys2 ideally because we dont do anything else for this subsystem here.
+        vars_to_solve = [*subsys2_outputs.keys()]
+        resd_to_solve = [*subsys2_residuals.keys()]
+
+        # total size of the jacobian
+        n_vars = 0
+        for var in vars_to_solve:
+            n_vars += subsys2_outputs[var].size
+
+        # initialize the schur complement jacobian for these variables
+        # TODO better way to get the dtype?
+        schur_jac = np.zeros((n_vars, n_vars),
```

Comment:
Since you're only using the system._vectors['residual']['linear'] vector array to determine the dtype, you shouldn't set `copy=True`